atomate/vasp/fireworks/core.py

```diff
@@ -131,232 +129,126 @@ def __init__(
+                new SCAN calculation using the provided structure.
+            prev_calc_dir (str): Path to a previous calculation to copy from
+            db_file (str): Path to file specifying db credentials.
+            parents (Firework): Parents of this particular Firework. FW or list of FWs.
+            vasptodb_kwargs (dict): kwargs to pass to VaspToDb
             **kwargs: Other kwargs that are passed to Firework.__init__.
         """
-        override_default_vasp_params = override_default_vasp_params or {}
-        orig_input_set = vasp_input_set or MPScanRelaxSet(
-            structure, **override_default_vasp_params
+        t = []
+
+        vasp_input_set_params = vasp_input_set_params or {}
+        vasptodb_kwargs = vasptodb_kwargs or {}
+        if "additional_fields" not in vasptodb_kwargs:
+            vasptodb_kwargs["additional_fields"] = {}
+        vasptodb_kwargs["additional_fields"]["task_label"] = name
+
+        fw_name = "{}-{}".format(
+            structure.composition.reduced_formula if structure else "unknown", name
         )
 
         # Raise a warning if the InputSet is not MPScanRelaxSet, because the
         # kspacing calculation from bandgap is only supported in MPScanRelaxSet.
-        if not isinstance(orig_input_set, MPScanRelaxSet):
+        if vasp_input_set and not isinstance(vasp_input_set, MPScanRelaxSet):
             raise UserWarning(
                 "You have specified a vasp_input_set other than \
-                               MPScanRelaxSet. Automatic adjustment of kspacing\
-                               is not supported by this InputSet."
+                 MPScanRelaxSet. Automatic adjustment of kspacing\
+                 is not supported by this InputSet."
             )
 
-        t = []
-        # write the VASP input files based on MPScanRelaxSet
-        t.append(WriteVaspFromIOSet(structure=structure, 
-                                    vasp_input_set=orig_input_set
-                                    )
-                 )
-
-        # pass the CalcLoc so that CopyFilesFromCalcLoc can find the directory
-        t.append(PassCalcLocs(name=name))
-
-        # Copy the pre-compiled VdW kernel for VASP, if required
-        if orig_input_set.vdw is not None:
-            t.append(CopyFiles(from_dir=vdw_kernel_dir))
+        if parents:
+            # Copy the CHGCAR from previous calc (usually PBE)
+            if prev_calc_dir:
+                t.append(CopyVaspOutputs(calc_dir=prev_calc_dir, contcar_to_poscar=True, additional_files=["CHGCAR"]))
+            elif prev_calc_loc:
+                t.append(CopyVaspOutputs(calc_loc=prev_calc_loc, contcar_to_poscar=True, additional_files=["CHGCAR"]))
+            else:
+                raise UserWarning("You specified parent Firework but did not provide its location. Set "
+                                  "prev_calc_dir or prev_calc_loc and try again.")
```

Comment:
@utf can you help me understand whether this is the correct way to pass a CalcLoc between two fireworks? Specifically, I'm confused about `prev_calc_loc`. The docstring for `CopyVaspOutputs` says

> 
> calc_loc (str OR bool): if True will set most recent calc_loc. If str
>             search for the most recent calc_loc with the matching name

Does that mean that if set to True, it will get the most recent calc_loc in the whole database? (i.e., whatever firework happens to have run just before this one) or the most recent calc_loc in the workflow?

I just want to make sure this code is robust and will always link the right fireworks together. Thanks!

Reply:
Hi @rkingsbury 

The idea is that there is a calc_locs list in the firework spec that can include the calculation location of the previous fireworks. This list is specific to a workflow rather than the whole database. The calc_locs list is appended by each firework that has finished running, and then passed to the next calculation. The latest calculation run will be the last calc loc in the list.

I think the logic here should follow the same process as the StaticFW. In particular, a firework may be initialised without parents but with `prev_calc_dir` set.

I've passed the logic of StaticFW below:

```python
        if prev_calc_dir:
            t.append(CopyVaspOutputs(calc_dir=prev_calc_dir, contcar_to_poscar=True))
            t.append(WriteVaspStaticFromPrev(other_params=vasp_input_set_params))
        elif parents:
            if prev_calc_loc:
                t.append(
                    CopyVaspOutputs(calc_loc=prev_calc_loc, contcar_to_poscar=True)
                )
            t.append(WriteVaspStaticFromPrev(other_params=vasp_input_set_params))
        elif structure:
            vasp_input_set = vasp_input_set or MPStaticSet(
                structure, **vasp_input_set_params
            )
            t.append(
                WriteVaspFromIOSet(structure=structure, vasp_input_set=vasp_input_set)
            )
        else:
            raise ValueError("Must specify structure or previous calculation")
```

To me, there is also a potential issue with the StaticFW logic: You could create a StaticFW with no parents and `prev_calc_loc=True` but then add parents added later (using `Worfklow.append_wf`). In that case, the previous calculation location will not be used.

IMO, it would be better to just throw a warning if `prev_calc_loc` is set but `parents=None`.

Reply:
Thanks @utf. I started with that logic but was trying to streamline it a bit, because there quite a few instructions that need to be executed when there is a previous calc, but not otherwise (in the above example, `t.append(WriteVaspStaticFromPrev(other_params=vasp_input_set_params))` is repeated; in my case there is more than that. 

I've updated my code to preserve the logic above, use a flag variable to avoid repeating all those instructions, and to raise a warning if `prev_calc_loc` is set but `parents` is not.